File: Singleton/Logger.js
class Logger{
    constructor(){
        this.logs=[];
    }
    static getLogger(){
        if(!Logger.instance){
            Logger.instance = new Logger();
        }
        return Logger.instance;
    }
    setData(d){
        this.logs.push(d);
    }
    getData(){
        return this.logs;
    }
}

const LoggerInstance = Logger.getLogger();
LoggerInstance.setData("hello World");
console.log(LoggerInstance.getData());

File: Singleton/Logger.java
import java.util.*;
class Singleton{ 
    private static Singleton instance;
    private ArrayList<String> logs;
    Singleton(){
        logs = new ArrayList<>();
    }
    public static Singleton getLogger(){
        if(instance == null){
            instance = new Singleton();
        }
        return instance;
    }
    public void setData(String s){
        logs.add(s);
    }
    public List getData(){
        return logs;
    }
}

class Logger{
    public static void main(String args[]){
        Singleton client = Singleton.getLogger();
        client.setData("hello world");
        System.out.println(client.getData());
        client.setData("Singleton example");
        System.out.println(client.getData());
    }
}
